tests/std/tests/P0896R4_views_join/test.cpp

```diff
@@ -0,0 +1,459 @@
+            range<input, Inner, Sized::yes, CanDifference::no, Common::no, CanCompare::no, ProxyRef::no>>();
+        Continuation::template call<Inner,
+            range<input, Inner, Sized::yes, CanDifference::no, Common::no, CanCompare::no, ProxyRef::yes>>();
+        Continuation::template call<Inner,
+            range<input, Inner, Sized::yes, CanDifference::no, Common::yes, CanCompare::yes, ProxyRef::no>>();
+        Continuation::template call<Inner,
+            range<input, Inner, Sized::yes, CanDifference::no, Common::yes, CanCompare::yes, ProxyRef::yes>>();
+
+        Continuation::template call<Inner,
+            range<input, Inner, Sized::yes, CanDifference::yes, Common::no, CanCompare::no, ProxyRef::no>>();
+        Continuation::template call<Inner,
+            range<input, Inner, Sized::yes, CanDifference::yes, Common::no, CanCompare::no, ProxyRef::yes>>();
+        Continuation::template call<Inner,
+            range<input, Inner, Sized::yes, CanDifference::yes, Common::yes, CanCompare::yes, ProxyRef::no>>();
+        Continuation::template call<Inner,
+            range<input, Inner, Sized::yes, CanDifference::yes, Common::yes, CanCompare::yes, ProxyRef::yes>>();
+
+        with_forward_ranges<Continuation, Inner>::template call<Inner>();
+    }
+};
+
+template <class Instantiator, class Element>
+constexpr void test_nested_inout() {
+    with_input_or_output_ranges<with_dependent_input_ranges<Instantiator>, Element>::call();
+}
+
+constexpr int expected_ints[]         = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+constexpr span<const int> intervals[] = {
+    {expected_ints + 0, expected_ints + 3},
+    {expected_ints + 3, expected_ints + 7},
+    {expected_ints + 7, expected_ints + 7},
+    {expected_ints + 7, expected_ints + 10},
+};
+
+struct instantiator {
+    template <ranges::range Inner, ranges::input_range Outer>
+    static constexpr void call() {
+        static_assert(ranges::size(intervals) == 4);
+        array inner_ranges = {Inner{intervals[0]}, Inner{intervals[1]}, Inner{intervals[2]}, Inner{intervals[3]}};
+        test_one(Outer{inner_ranges}, expected_ints);
+    }
+};
+
+using mo_inner = test::range<input_iterator_tag, const int, test::Sized::no, test::CanDifference::no, test::Common::no,
+    test::CanCompare::no, test::ProxyRef::yes, test::CanView::yes, test::Copyability::copyable>;
+
+template <class Category, test::Common IsCommon, bool is_random = derived_from<Category, random_access_iterator_tag>>
+using move_only_view = test::range<Category, mo_inner, test::Sized{is_random}, test::CanDifference{is_random}, IsCommon,
+    test::CanCompare{derived_from<Category, forward_iterator_tag>}, test::ProxyRef::no, test::CanView::yes,
+    test::Copyability::move_only>;
+void test_move_only_views() {
+    const auto gen = [] {
+        return array{mo_inner{intervals[0]}, mo_inner{intervals[1]}, mo_inner{intervals[2]}, mo_inner{intervals[3]}};
+    };
+
+    auto data = gen();

```

Comment:
This test is `using namespace std;` so I recommend avoiding `data`. Occurs in `main` below.